code/game/objects/items/kitchen.dm

```diff
@@ -55,6 +56,30 @@
 	else
 		return ..()
 
+/obj/item/kitchen/knife/poison
+	name = "venom knife"
+	icon_state = "poisonknife"
+	force = 12
+	throwforce = 15
+	throw_speed = 5
+	throw_range = 7
+	desc = "An infamous knife of syndicate design,it has a tiny hole going through the blade to the handle which stores toxins."
+	materials = null
+
+/obj/item/kitchen/knife/poison/Initialize()
+	. = ..()
+	create_reagents(40,OPENCONTAINER)
+
+/obj/item/kitchen/knife/poison/attack(mob/living/M, mob/user)
+	if (!istype(M))
+		return
+	. = ..()
+	if (!reagents.total_volume || !M.reagents)
+		return
+	var/randvar = pick(3,5)
+	var/amount = min(randvar/reagents.total_volume,1)

```

Comment:
Can you explain your thought process for the math here?

Reply:
>ports the poison knife from citadelv
🤷‍♂

Reply:
although i see that line 80 is redundant

Reply:
Well what is it meant to do

Reply:
well, var/randvar picks 3 4 or 5 units.(90% sure that (3,5) is 3-5 not 3 or 5))
var/amount is the inject amount which is decided by how much ranvar picks. i guess
still man reeee merge it i tested it, it works no runtimes no poo shit cum its great now merje.

Reply:
But why does it divide by reagents.total_volume